### PR TITLE
FaultInjection: Fix random number generation

### PR DESCRIPTION
**Pull request by yashykt**
`rand()` is not thread-safe, so instead use `absl::InsecureBitGen` and `absl::Uniform` to generate random numbers.

### CODE REVIEW

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc [hunk 1]
```diff
@@ -204,7 +204,13 @@ TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageAbortViaHeaders) {
 
 TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageDelay) {
   CreateAndStartBackends(1);
+  const uint32_t kFixedDelayMilliseconds = 100000;
+#ifndef NDEBUG
```
Comment (markdroth) [inline test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:208]: I'm really not keen on putting `#ifndef`s in the middle of individual tests.  Can we not just increase the timeout even for opt builds?

**Review by markdroth — COMMENTED** [review]
This looks really good!  Just one comment to address.

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc (hunk 1 above)
Comment (yashykt) [reply test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc]: Done

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc [hunk 2]
```diff
@@ -204,11 +204,11 @@ TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageAbortViaHeaders) {
 
 TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageDelay) {
   CreateAndStartBackends(1);
-  const uint32_t kRpcTimeoutMilliseconds = grpc_test_slowdown_factor() * 3000;
+  const uint32_t kRpcTimeoutMilliseconds = 10000;
```
Comment (markdroth) [inline test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:207]: Don't we still want to use `grpc_test_slowdown_factor()` here, with whatever timeout value we choose?

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc [hunk 3]
```diff
@@ -250,11 +250,11 @@ TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageDelay) {
 TEST_P(FaultInjectionTest, XdsFaultInjectionPercentageDelayViaHeaders) {
   CreateAndStartBackends(1);
   const uint32_t kFixedDelayMilliseconds = 100000;
-  const uint32_t kRpcTimeoutMilliseconds = grpc_test_slowdown_factor() * 3000;
+  const uint32_t kRpcTimeoutMilliseconds = 10000;
```
Comment (markdroth) [inline test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:253]: Same here.

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc (hunk 2 above)
Comment (yashykt) [reply test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:207]: This is used in RpcOptions, and RpcOptions has it's own grpc_test_slowdown_factor() multiplier -
https://github.com/grpc/grpc/blob/2a7286a67c71e2e0c0eee5e49ba1a719fcabb3a8/test/cpp/end2end/xds/xds_end2end_test_lib.cc#L419
https://github.com/grpc/grpc/blob/2a7286a67c71e2e0c0eee5e49ba1a719fcabb3a8/test/core/util/test_config.cc#L81

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc (hunk 3 above)
Comment (yashykt) [reply test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:253]: see above

File: test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc (hunk 2 above)
Comment (markdroth) [reply test/cpp/end2end/xds/xds_fault_injection_end2end_test.cc:207]: Ah, good catch!

**Review by markdroth — APPROVED** [review]

**Comment by yashykt** [comment]
Thanks for reviewing!

**Pull request merged**